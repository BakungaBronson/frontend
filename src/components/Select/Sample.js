import React from 'react';
import Select from './index';

const SelectSample = () => (
  <div style={{ backgroundColor: '#f1f1f1', padding: '50px', width: '400px' }}>
    <Select
      required
      placeholder="Choose a location"
    />
  </div>
);

export default SelectSample;
