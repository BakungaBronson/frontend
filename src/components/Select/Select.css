.SelectWrapper {
  width: 100%;
}

.SelectElementMain {
  height: 2rem;
  border: 2px solid black;
  padding-top: 0.5rem;
  padding-left: 0.5rem;
}