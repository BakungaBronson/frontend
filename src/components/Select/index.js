import React, { useState } from 'react';
import './Select.css';

const Select = ({ required, placeholder }) => {
  const [selectValue, setValue] = useState(`${placeholder} ${required ? '*' : ''}`);

  return (
    <div className="SelectWrapper">
      <div className="SelectElementMain">
        {selectValue}
      </div>
      <div className="SelectOptionsWrapper"></div>
    </div>
  );
};

export default Select;
