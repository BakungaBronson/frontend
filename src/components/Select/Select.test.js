import React from 'react';

import { shallow } from 'enzyme';

import Select from './index';

describe('Select Component', () => {
  it('Renders a Select component', () => {

    const selectEvent = jest.fn();
    const SelectComponent = shallow(<Select onChange={selectEvent} options={[{"id": 1, "name":"test"}]} required placeholder='Testing'/>);
    expect(SelectComponent).toMatchSnapshot();

  });
});