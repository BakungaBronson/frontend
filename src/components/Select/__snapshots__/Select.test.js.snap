// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Renders a Select component 1`] = `ShallowWrapper {}`;
